example/main.go

```diff
@@ -37,21 +39,51 @@ func main() {
 		},
 	}
 
-	s := mcp.NewServer(serverInfo, tools)
+	prompts := []mcp.PromptDefinition{
+		{
+			Metadata: mcp.Prompt{
+				Name:        "example",
+				Description: ptr("An example prompt template"),
+				Arguments: []mcp.PromptArgument{
+					{
+						Name:        "text",
+						Description: ptr("Text to process"),
+						Required:    ptr(true),
+					},
+				},
+			},
+			Process:   processPrompt,
+			RateLimit: rate.NewLimiter(rate.Every(time.Second), 5),
```

Comment:
The spec doesn't require rate limiting of prompt gets in the same way that it does for tool calls. However having a rate limit seems reasonable.